account_invoice_supplierinfo_update_standard_price/models/account_invoice_line.py

```diff
@@ -22,8 +23,15 @@ def _get_standard_price(self):
                 factor =\
                     self.uos_id.factor_inv\
                     / self.product_id.uom_id.factor_inv
-            line_shared_cost = self.invoice_id.distributed_expense_total * (
-                self.price_subtotal / self.invoice_id.product_expense_total)
+            if self.invoice_id.product_expense_total != 0:
+                line_shared_cost =\
+                    self.invoice_id.distributed_expense_total * (
+                        self.price_subtotal /
+                        self.invoice_id.product_expense_total)
+            else:
+                raise UserError(_(
+                    "We can't check prices\
```

Comment:
"We can't check prices"
                     " for a invoice whose total is null."))

Reply:
ok c'est bon :)